frame/staking/src/pallet/mod.rs

```diff
@@ -559,6 +651,184 @@ pub mod pallet {
 	#[pallet::getter(fn current_planned_session)]
 	pub type CurrentPlannedSession<T> = StorageValue<_, SessionIndex, ValueQuery>;
 
+	/// Wrapper struct for Era related information. It is not a pure encapsulation as these storage
+	/// items can be accessed directly but nevertheless, its recommended to use `EraInfo` where we
+	/// can and add more functions to it as needed.
+	pub(crate) struct EraInfo<T>(sp_std::marker::PhantomData<T>);
+	impl<T: Config> EraInfo<T> {
+		/// Temporary function which looks at both (1) passed param `T::StakingLedger` for legacy
+		/// non-paged rewards, and (2) `T::ClaimedRewards` for paged rewards. This function can be
+		/// removed once `$HistoryDepth` eras have passed and none of the older non-paged rewards
+		/// are relevant/claimable.
+		// Refer tracker issue for cleanup: #13034
+		pub(crate) fn is_rewards_claimed_temp(
+			era: EraIndex,
+			ledger: &StakingLedger<T>,
+			validator: &T::AccountId,
+			page: PageIndex,
+		) -> bool {
+			ledger.legacy_claimed_rewards.binary_search(&era).is_ok() ||
+				Self::is_rewards_claimed(era, validator, page)
+		}
+
+		/// Check if the rewards for the given era and page index have been claimed.
+		///
+		/// This is only used for paged rewards. Once older non-paged rewards are no longer
+		/// relevant, `is_rewards_claimed_temp` can be removed and this function can be made public.
+		fn is_rewards_claimed(era: EraIndex, validator: &T::AccountId, page: PageIndex) -> bool {
+			ClaimedRewards::<T>::get(era, validator).binary_search(&page).is_ok()
```

Comment:
I was keeping it sorted to find the next claimable page easily but since this vec is gonna be small, I believe I can just loop through the numbers till the max page count.